bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthClientServiceImpl.java

```diff
@@ -333,7 +333,7 @@ public AccessTokenResponse refreshToken() throws OAuthException, IOException, OA
     }
 
     @Override
-    public @Nullable AccessTokenResponse getAccessTokenResponse()
+    public synchronized @Nullable AccessTokenResponse getAccessTokenResponse()
```

Comment:
Is this really needed when having `refreshToken()` synchronized?

Reply:
> Is this really needed when having refreshToken() synchronized?

Yes I think do. If we just rely on the refreshToken sync, I can imagine that a second call to this method might still retrieve the old token before refreshToken has completed and stored the new one .. and then it would have to call refreshToken again..

Reply:
Agree with @andrewfg